### PR TITLE
feat: os version analytics

### PR DESCRIPTION
**Pull request by robertbrada**
Added analytics to track OS version of the desktop app

### CODE REVIEW

**Comment by keraf** [comment]
I know it's not final but just I would suggest using the `app/ready` method to set certain values in the UI that can be fetched at launch and will never update while the app is running. OS and version is a good candidate for that :slightly_smiling_face: 

Edit: Just checked the source, looks like `app/ready` has been removed. In any case, I think it would be a good idea to add it back and use it in the future for any constant values we want to set in the UI from the electron application.

**Comment by robertbrada** [comment]
> I know it's not final but just I would suggest using the `app/ready` method to set certain values in the UI that can be fetched at launch and will never update while the app is running. OS and version is a good candidate for that 
> 
> Edit: Just checked the source, looks like `app/ready` has been removed. In any case, I think it would be a good idea to add it back and use it in the future for any constant values we want to set in the UI from the electron application.

Initializing `analytics()` in `app.on('ready', init)` is not what you meant by using `app/ready` function? Or is it?

**Comment by mroz22** [comment]
Raf meant that os data that you are sending now through analytics/something channel could be also sent in app/ready event. This event is not around at the moment. But if it was, you could send it and save os information into application state and have free access to using it any time later.

**Comment by robertbrada** [comment]
Alright I will try to do it that way if it's a better approach

**Comment by mroz22** [comment]
For your case not really better, but it might be useful in future. I am ok with your current implementation (maybe wil find few nitpicks).

File: packages/suite/src/actions/suite/analyticsActions.ts [hunk 1]
```diff
@@ -63,6 +63,7 @@ export type AnalyticsEvent =
               theme: string;
               // added in 1.6
               suiteVersion: string;
+              desktopOSVersion: { platform: string; release: string } | '';
```
Comment (mroz22) [inline packages/suite/src/actions/suite/analyticsActions.ts]: not sure here. how does it look like serialized? If I remember correctly, properties on `payload` object were always primitives till now.

File: packages/suite/src/middlewares/suite/analyticsMiddleware.ts [hunk 2]
```diff
@@ -32,7 +33,17 @@ const analytics = (api: MiddlewareAPI<Dispatch, AppState>) => (next: Dispatch) =
     const state = api.getState();
 
     switch (action.type) {
-        case STORAGE.LOADED:
+        case STORAGE.LOADED: {
+            // get OS version of the desktop app (for web returns resp:undefined)
+            let desktopOSVersion;
+            const resp = await getOSVersion();
```
Comment (mroz22) [inline packages/suite/src/middlewares/suite/analyticsMiddleware.ts]: this works for web? it looks kind of misfit here. let me think about it

File: packages/suite/src/actions/suite/analyticsActions.ts (hunk 1 above)
Comment (robertbrada) [reply packages/suite/src/actions/suite/analyticsActions.ts]: Yes, all properties in the payload are primitive till now. Maybe sending string in a format `${platform}_${release}` would be better?

File: packages/suite/src/middlewares/suite/analyticsMiddleware.ts (hunk 2 above)
Comment (robertbrada) [reply packages/suite/src/middlewares/suite/analyticsMiddleware.ts]: For web no particular error is thrown, but `await getOSVersion()` return `undefined`. But  there might be a better way how to detect if the suite is running on desktop

Comment (mroz22) [reply packages/suite/src/middlewares/suite/analyticsMiddleware.ts]: you also don't need necessarily to send this in information in suite-ready event but you can create a new analytics event and dispatch it in the end of analyticsActions.init

Comment (robertbrada) [reply packages/suite/src/middlewares/suite/analyticsMiddleware.ts]: Yes maybe creating a custom action that would send `desktopOSVersion` analytics only if desktop Suite is being used would be better

File: packages/suite/src/middlewares/suite/analyticsMiddleware.ts [hunk 3]
```diff
@@ -32,7 +32,7 @@ const analytics = (api: MiddlewareAPI<Dispatch, AppState>) => (next: Dispatch) =
     const state = api.getState();
 
     switch (action.type) {
-        case STORAGE.LOADED:
+        case STORAGE.LOADED: {
```
Comment (tomexx) [inline packages/suite/src/middlewares/suite/analyticsMiddleware.ts]: are `{}` needed?

File: packages/suite/src/utils/suite/env.ts [hunk 4]
```diff
@@ -115,3 +115,8 @@ export const getOSTheme = (): SuiteThemeVariant => {
     }
     return 'light';
 };
+
+export const getOSVersion = async () => {
+    const resp = await window.desktopApi?.getOSVersion();
```
Comment (tomexx) [inline packages/suite/src/utils/suite/env.ts]: it can be returned directly

**Review by tomexx — COMMENTED** [review]
nitpicking

File: packages/suite/src/middlewares/suite/analyticsMiddleware.ts [hunk 5]
```diff
@@ -58,6 +58,7 @@ const analytics = (api: MiddlewareAPI<Dispatch, AppState>) => (next: Dispatch) =
                 }),
             );
             break;
+
```
Comment (keraf) [inline packages/suite/src/middlewares/suite/analyticsMiddleware.ts]: Rogue space

File: packages/suite-desktop/src-electron/libs/constants.ts [hunk 6]
```diff
@@ -21,4 +21,6 @@ export const MODULES = [
     { name: 'auto-updater', dependencies: ['mainWindow', 'store'], isDev: false },
     // Dev Only
     { name: 'dev-tools', dependencies: ['mainWindow'], isDev: true },
+    // Analytics
+    { name: 'analytics', dependencies: [] },
```
Comment (keraf) [inline packages/suite-desktop/src-electron/libs/constants.ts]: This one can be moved a bit higher up with the Standard modules.

**Review by keraf — APPROVED** [review]
Very minor nitpick. The rest looks good. I guess a quick fixup and it's ready to go :slightly_smiling_face:

**Pull request merged**